### PR TITLE
[core][autoscaler] Configurable event level logging to driver from autoscaler for autoscaler event in V2. 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
This allows users to configure the event level to be streamed to driver by the env var. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/event/event_logger.py [hunk 1]
```diff
@@ -157,3 +163,33 @@ def parse_event(event_str: str) -> Optional[Event]:
...
+    Args:
+        event: The event to filter.
+        filter_event_level: The event level string to filter by. Any events
+            that are lower than this level will be filtered.
+
+    Returns:
+        True if the event should be filtered, else False.
+    """
+
+    event_levels = {
```
Comment (rkooo567) [inline python/ray/_private/event/event_logger.py:180]: can you assert fail the non-existent level?

File: python/ray/_private/ray_constants.py [hunk 2]
```diff
@@ -39,6 +39,9 @@ def env_set_by_user(key):
 # Whether event logging to driver is enabled. Set to 0 to disable.
 AUTOSCALER_EVENTS = env_integer("RAY_SCHEDULER_EVENTS", 1)
 
+# Filter level under which events will be filtered out, i.e. not printing to driver
+RAY_LOG_TO_DRIVER_EVENT_LEVEL = os.environ.get("RAY_LOG_TO_DRIVER_EVENT_LEVEL", "INFO")
```
Comment (rkooo567) [inline python/ray/_private/ray_constants.py:43]: Can we do WARNING by default? Or does it break existing legacy autoscaler logs?

**Review by rkooo567 — APPROVED** [review]
If INFO is necessray, can we make sure the autoscaler logs don't regress with the prod autoscaler? I.e., we shouldn't print more information than we print to the application logs now.

File: python/ray/_private/ray_constants.py (hunk 2 above)
Comment (rickyyx) [reply python/ray/_private/ray_constants.py:43]: legacy is INFO.

Comment (rickyyx) [reply python/ray/_private/ray_constants.py:43]: but we could also change the leagcy level.

**Comment by rickyyx** [comment]
> If INFO is necessray, can we make sure the autoscaler logs don't regress with the prod autoscaler? I.e., we shouldn't print more information than we print to the application logs now.

cc @hanming-lu @shomilj

File: python/ray/_private/event/event_logger.py (hunk 1 above)
Comment (rickyyx) [reply python/ray/_private/event/event_logger.py:180]: `Event.Severity.Value(lvl)` should raise an error for the non-existent level.

Comment (shomilj) [reply python/ray/_private/event/event_logger.py:180]: Hmm, are you sure you want to fail non-existent levels? This goes against `.proto` best practices & makes version compatibility harder.

Comment (rickyyx) [reply python/ray/_private/event/event_logger.py:180]: i guess the assertions was more on the supplied filter level - which could be not a valid level.

**Pull request merged**